src/cpp/src/llm_pipeline_base.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/llm_pipeline.hpp"
+#include "openvino/genai/tokenizer.hpp"
+#include "openvino/genai/streamer_base.hpp"
+
+namespace ov {
+namespace genai {
+
+class LLMPipelineImplBase {
+public:
+    LLMPipelineImplBase(const Tokenizer& tokenizer,
+                        const GenerationConfig& config = {})
+    : m_tokenizer(tokenizer), m_generation_config(config) {
+    }
+
+    virtual DecodedResults generate(
+        StringInputs inputs,
+        OptionalGenerationConfig generation_config,
+        StreamerVariant streamer
+    ) = 0;
+
+    virtual EncodedResults generate(
+        const EncodedInputs& inputs,
+        OptionalGenerationConfig generation_config,
+        StreamerVariant streamer
+    ) = 0;
+
+    virtual void start_chat() = 0;
```

Comment:
as I see NPU pipeline does not implement this method. Is it OK ?
at least to throw exception that NPU does not support chat mode for now.

Reply:
NPU does implement it, otherwise it won't be possible to instantiate `NPULLMPipelineImpl`:
https://github.com/TolyaTalamanov/openvino.genai/blob/at/integrate-npu-friendly-llm-execution/src/cpp/src/llm_pipeline_npu.hpp#L36-L41